WcaOnRails/app/webpacker/components/UserBadge.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  Button, Icon, Label, Popup,
+} from 'semantic-ui-react';
+import UserAvatar from './UserAvatar';
+import I18n from '../lib/i18n';
+
+import '../stylesheets/user_badge.scss';
+
+/**
+ * @param {{ leader: boolean, senior_member: boolean}} user
+ * @returns {string[]}
+ */
+export function subtextForMember(user) {
+  if (user.leader) {
+    return [I18n.t('about.structure.leader')];
+  }
+  if (user.senior_member) {
+    return [I18n.t('about.structure.senior_member')];
+  }
+
+  return [];
+}
+
+/**
+ * @param {{ teams: { friendly_id: string, leader: boolean }[]}} user
+ * @param {{ friendly_id: string, name: string }[]} officerTitles
+ * @returns {string[]}
+ */
+export function subtextForOfficer(user, officerTitles) {
+  const positions = user.teams
+    .map((team) => {
+      const title = officerTitles.find((t) => t.friendly_id === team.friendly_id);
+      if (title) return title.name;
```

Comment:
nit: Once again, ternary operators can clean this code up.
Or even just a

```js
return title ? title.name : ''
```

Or if this repository supports a slightly more advance js feature:

```js
return title?.name || '';
```

Reply:
Yep fair enough; unfortunately optional chaining does not seem to be supported...